shuffle/genesis/tests/e2e.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use diem_config::config::NodeConfig;
+use diem_crypto::PrivateKey;
+use diem_sdk::{
+    client::BlockingClient,
+    transaction_builder::{
+        stdlib::encode_create_parent_vasp_account_script_function, Currency, TransactionFactory,
+    },
+    types::LocalAccount,
+};
+use diem_types::{
+    account_config, chain_id::ChainId, on_chain_config::VMPublishingOption,
+    transaction::authenticator::AuthenticationKey,
+};
+
+#[test]
+fn node_e2e() -> Result<()> {
+    let node_config_dir = diem_temppath::TempPath::new();
+    let validator_config = shuffle_custom_node::generate_validator_config(
+        node_config_dir.path(),
+        VMPublishingOption::open(),
+    )?;
+    let node_config = NodeConfig::load(validator_config.config_path())?;
+    let root_key = generate_key::load_key(node_config_dir.path().join("mint.key"));
+
+    let json_rpc_url = format!("http://0.0.0.0:{}", node_config.json_rpc.address.port());
+    // Start the node in a separate thread using our config
+    std::thread::spawn(move || {
+        diem_node::start(&node_config, None);
+    });
+    // Wait for node to start.
+    std::thread::sleep(std::time::Duration::from_secs(1));
```

Comment:
There are a few things to unpack here, with both the starting of the node for the e2e test (L32), and the sleeping until it's ready (L35). Forge, which was written to facilitate e2e tests, can't accommodate our current use case, so we'll be sticking with this approach. At the very least though, could we capture the subtle goal here of having the parent test process kill off the node? Comments and perhaps a documenting variable would go a long way:

```
    let _thread_killed_by_test = std::thread::spawn(move || {
        diem_node::start(&node_config, None);
    });
```